### ISSUE TITLE
Pass JavaScript to Component

### ISSUE THREAD

**Issue by JasonCaterino** [vmware-tanzu/octant#695]
**Describe the problem/challenge you have**
I am part of a group of college students working with AT&T on Airship2.0
We are wanting to import a Javascript graph library to create a YAML visualizer plug-in for Airship UI. We are going to need to create our own HTML, JavaScript, and Go Octant Plugin.
Currently Octant does not allow us to pass resources from a plugin to Octant (as in saved JavaScript libraries). This limits what our plugins can do since we would have to add anything new as a resource in Octant .

**Describe the solution you'd like**
We would like to be able to create our own HTML elements and import our own Javascript libraries.

**Comment by wwitzel3** [comment]
I know @bryanl did a proof of concept to allow just this feature for plugins a while back. We can look in to resurrecting that and turning it in to a feature for plugins.

**Comment by aschiefe** [comment]
+1 on this

**Comment by RichiCoder1** [comment]
It may be a little overly complicated, but maybe somewhere where you could take inspiration from [Jira Cloud](https://developer.atlassian.com/cloud/jira/software/)?

In essence, the plugin provides a descriptor with a URL. Octant uses that descriptor to load a seamless iframe in the appropriate place with the appropriate restrictions. The iframe can use [a simple javascript API provided by a provided script](https://developer.atlassian.com/cloud/jira/platform/about-the-javascript-api/) to commicate with Octant client side (and obviously would just use normal plugin methods to talk server side). Some great tools like [simple-xdm](https://bitbucket.org/atlassian/simple-xdm/src/master/) by atlassian and [post-robot](https://github.com/krakenjs/post-robot) by paypal to keep that interaction simple.

**Comment by aschiefe** [comment]
It's a good idea and I think it'll work in some if not most situations, and I believe that octant already has an [iframe](https://github.com/vmware-tanzu/octant/blob/master/pkg/view/component/iframe.go) component to it.  Iframes have some inherit problems when you talk about poking data through a more complex network structure, and that's a situation we have on our side pretty frequently.  When your octant is on a remote machine trying to pass data your local machine tends to cause CORS type issues where the iframe won't render and you may not have access to the serving components to add the correct headers.  Just some thoughts on iframe, though again it would probably work for most people, just not for the topology that we're trying to solve for.  Having the ability to put up your own web components would very much help for our particular situaiton.

**Comment by bryanl** [comment]
@RichiCoder1 correct, you can do this, and this is the feature that I prototyped a while back. There should be some movement on this functionality soon as I personally need it to scratch an itch I have.
